contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -437,7 +437,7 @@ func (q *queuer) Run(done <-chan struct{}) {
 			}
 
 			pod := p.(*Pod)
-			if recoverAssignedSlave(pod.Pod) != "" {
+			if recoverAssignedSlave(pod.Pod) != "" || pod.InGracefulTermination() {
```

Comment:
The apiserver checks the NodeName. If it is empty, the deletion is immediate. If it is set, the graceful timeout will apply. In the case the NodeName is set to a non-existing node, the pod will countdown its graceful deletion timeout even in upstream.

We have the special case that we schedule pods with NodeName != "". Even though this case only happens for pre-scheduled pods which are gracefully deleted, I would leave the above predicate. I will add a comment.